extensions/mssql/src/hdfs/aclEntry.ts

```diff
@@ -135,12 +137,42 @@ export class AclEntry {
 	 * Example strings :
 	 *		user:bob:rwx
 	 *		default:user:bob:rwx
-	 * 		user::r-x
-	 * 		default:group::r--
+	 *		user::r-x
+	 *		default:group::r--
 	 */
-	toString(): string {
-		return `${this.scope === AclEntryScope.default ? 'default:' : ''}${this.type}:${this.name}:${this.permission.toString()}`;
+	toAclString(): string {
+		return `${this.scope === AclEntryScope.default ? 'default:' : ''}${getAclEntryType(this.type)}:${this.name}:${this.permission.toString()}`;
 	}
+
+	/**
+	 * Checks whether this and the specified AclEntry are equal. Two entries are considered equal
+	 * if their scope, type and name are equal.
+	 * @param other The other entry to compare against
+	 */
+	public isEqual(other: AclEntry): boolean {
+		if (!other) {
+			return false;
+		}
+		return this.scope === other.scope &&
+			this.type === other.type &&
+			this.name === other.name;
+	}
+}
+
+/**
+ * Maps the possible entry types into their corresponding values for using in an ACL string
+ * @param type The type to convert
+ */
+function getAclEntryType(type: AclEntryType | AclPermissionType): AclEntryType {
+	switch (type) {
+		case AclPermissionType.owner:
+			return AclEntryType.user;
+		case AclPermissionType.group:
+			return AclEntryType.group;
+		case AclPermissionType.other:
+			return AclEntryType.other;
```

Comment:
can we add a default case and throw error here? easier to catch if we add new options.